tests/func/test_query.py

```diff
@@ -194,11 +189,12 @@ def test_query(
     """
     query_script = setup_catalog(query_script, catalog_info_filepath)
 
-    result = catalog.query(query_script, save=save, save_as=save_as)
-    if save_as:
-        assert result.dataset.name == save_as
-        assert catalog.get_dataset(save_as)
-    elif save_dataset:
+    result = catalog.query(query_script, save=save)
+    if not save:
+        assert result.dataset is None
+        return
```

Comment:
Previous test did not test with `save=False` and when `save_as=None`. I am testing this just to make parametrization easier. #360 will require adjusting these anyway.